modules/javafx.graphics/src/main/java/javafx/scene/text/Text.java

```diff
@@ -1821,6 +1877,36 @@ final BooleanProperty caretBiasProperty() {
             }
             return caretBias;
         }
+
+        private IntegerProperty tabSize;
+
+        final int getTabSize() {
+            return tabSize == null ? TextLayout.DEFAULT_TAB_SIZE : tabSize.get();
+        }
+
+        final IntegerProperty tabSizeProperty() {
+            if (tabSize == null) {
+                tabSize = new StyleableIntegerProperty(TextLayout.DEFAULT_TAB_SIZE) {
+                    @Override public Object getBean() { return Text.this; }
+                    @Override public String getName() { return "tabSize"; }
+                    @Override public CssMetaData getCssMetaData() {
+                        return StyleableProperties.TAB_SIZE;
+                    }
+                    @Override public void set(int v) { super.set((v < 1) ? 1 : v); }
```

Comment:
For mutable properties, we usually clamp on usage, so that we don't have problems binding to the value. This preserves the invariant that `set(val); get() == val` for all values. If that is what we end up doing, then this overridden method should be removed.

Reply:
Hmm ... so you are saying the clamping is the responsibility of client code (internal as well as external)?

Reply:
In this case, yes.